index.js

```diff
@@ -0,0 +1,106 @@
+// for each number click, concatnate the value string of that number to the current operant
+// maybe use .innerHtml or something reaserch what you can use
+// then when an operant is clicked, cuurent operasant is now previous operant and
+// then modfiy current operant to be "". also save the operant that was clicked on a varaible
+// then when numbers are clicked, it should be concatnated to currentOperant.
+// then when equalSButton  is clicked, we done a switch staement to check which operant is used and
+// use corresponding function on calcualtor to calculate previousopprant, currentOperant with the operant
+// then call .getTotal()
+
+const screen = document.querySelector("#screen");
+const numbers = document.querySelectorAll("[numbersToPress]");
+const operationButtons = document.querySelectorAll(".operationButtons");
+const equalsButton = document.querySelector("#equalsButton");
+const clearButton = document.querySelector("#clearButton");
+const historyButton = document.querySelector("#getHistory");
+const decimalButton = document.querySelector("#decimalButton");
+let secondValue = "";
+let operant = "";
+let currentValue = 0;
+let operantPressed = 0;
+
+const calculator = new Calculator();
+
+function returnValueOfOperation() {
+  if (screen.value === "" || screen.value === ".") return;
+
+  if (secondValue.includes(".")) {
+    currentValue = parseFloat(secondValue);
+  } else {
+    currentValue = parseInt(secondValue);
+  }
+
+  switch (operant) {
+    case "+":
+      calculator.add(currentValue);
+      screen.value = calculator.getTotal();
+      break;
+    case "-":
+      calculator.subtract(currentValue);
+      screen.value = calculator.getTotal();
+      break;
+    case "*":
+      calculator.multiply(currentValue);
+      screen.value = calculator.getTotal();
+      break;
+    case "/":
+      calculator.divide(currentValue);
+      screen.value = calculator.getTotal();
+      break;
+    case "^":
+      calculator.power(currentValue);
+      screen.value = calculator.getTotal();
+      break;
+    default:
+      return currentValue;
+  }
+}
+
+clearButton.addEventListener("click", () => {
+  screen.value = "";
+  secondValue = "";
+  operationClicked = false;
+  operantPressed = 0;
+  calculator.clear();
+});
+
+numbers.forEach((button) => {
+  button.addEventListener("click", () => {
+    if (operantPressed) {
+      secondValue += button.innerText;
+      screen.value = button.innerText;
+      operantPressed--;
+    } else {
+      screen.value = screen.value + button.innerText;
+      secondValue += button.innerText;
+    }
+  });
+});
+
+operationButtons.forEach((button) => {
+  button.addEventListener("click", () => {
+    if (screen.value === "") return;
+    operantPressed = true;
+    calculator.total = parseFloat(screen.value.toString());
+    operant = button.innerText;
+    secondValue = "";
+  });
+});
+
+decimalButton.addEventListener("click", () => {
```

Comment:
Why in this function arr you checking the decimalButton's innerText value? Isn't it always "." anyway? I think this is old code.

Reply:
In fact this function should be simplified greatly, most of the logic is very strange and meaningless at the moment

Reply:
> d

True, forgot about that. Changed it.

Reply:
I took out the part you told me about. But to me it seems like its now simplified to the best it can. I am not allowing for multiple "." to be shown on screen value, and if someone pressses the "." first, then the screen value will be "0." which logically makes sense. Or is that what you meant by currently it is meaningless?